IE-356 + Added log retrieval command

dspasojevic opened the pull request:
Command line options:

```
Retrieves logs from okta

Okta connection settings:
      --client-id, --cid                   Okta client ID    [string] [required]
      --private-key, --pk                  Okta private key as string form of
                                           JSON              [string] [required]
      --organisation-url, --org-url, --ou  Okta URL for Organisation  [required]

Options:
      --version        Show version number                             [boolean]
      --help           Show help                                       [boolean]
  -o, --output-format  The format to output the logs in.
                          [string] [choices: "json", "table"] [default: "table"]
  -l, --limit          The number of logs to retrieve.    [number] [default: 10]
  -q, --query          The query by which the logs will be filtered (e.g. OIDC).
                                                                        [string]
  -f, --filter         The filter by which the logs will be filtered (e.g.
                       eventType eq "user.session.start").              [string]
      --since          The start date of the logs to retrieve.          [string]
      --until          The end date of the logs to retrieve.            [string]
```

Same output in table mode:
```
 ID                                    Published                 Severity  Type                Who    Message
 649c367a-9306-11ee-906b-9f37ae57e05c  2023-12-05T00:36:56.211Z  WARN      user.session.start  Dan S  User login to Okta
 c21c865c-9306-11ee-82fb-9b4e2ab3682d  2023-12-05T00:39:33.080Z  WARN      user.session.start  Dan S  User login to Okta
```

desimpson reviewed (commented):
I noticed [this build error](https://github.com/agiledigital-labs/oktagon/actions/runs/7094589459/job/19310113266?pr=48) relating to some linting errors:
```
   102:7  error  Unused eslint-disable directive (no problems were reported from 'functional/no-return-void')
  104:9  error  Unused eslint-disable directive (no problems were reported from 'functional/no-let')
```

desimpson commented on src/scripts/services/okta-service.ts:
```diff
@@ -74,6 +75,55 @@ export const validateCredentials = (
+    TE.tryCatch(
+      // eslint-disable-next-line functional/no-return-void
+      async () => {
+        // eslint-disable-next-line functional/no-let, functional/prefer-readonly-type
```
I think this `functional/no-let` may have been left over from different code?

dspasojevic replied:
It is. Fixed.

desimpson commented on src/index.ts:
```diff
@@ -70,6 +65,12 @@ const rootCommand = yargs
   .help();
 
+/**
+ * Dynamic type for global arguments. This needs to be it's own as we use a
```
Typo from previous code: it's -> its.

desimpson reviewed (changes requested):
Looks great.

dspasojevic commented:
Thanks @desimpson. I've addressed your comments - this it ready for re-review.

As an aside, we are switching to https://conventionalcomments.org/ so that we have a shared understanding about what is required (e.g. is something a blocking issue or a suggestion) to complete the review.

desimpson commented:
> Thanks @desimpson. I've addressed your comments - this it ready for re-review.
> 
> As an aside, we are switching to https://conventionalcomments.org/ so that we have a shared understanding about what is required (e.g. is something a blocking issue or a suggestion) to complete the review.

Thanks, I'll use these labels in future.

desimpson reviewed (approved):
LGTM.
